search_engine_parser/core/engines/google.py

```diff
@@ -87,7 +92,19 @@ def parse_single_result(self, single_result, return_type=ReturnType.FULL, **kwar
                 link_tag = desc_tag.find('a')
                 if link_tag:
                     raw_link = link_tag.get('href')
+                    results['hrefs'] = self.extract_href(raw_link)
                     results['links'] = self.parse_url(raw_link)
             desc = desc_tag.text
             results['descriptions'] = desc
         return results
+
+    def extract_href(self, url):
+        """Overwrite BaseSearch::extract_href."""
+        parsed = urlparse.urlparse(url)
+        url_qs = parse_qs(parsed.query)
+        if 'q' in url_qs:
+            return url_qs['q']
```

Comment:
Add unquote at the top
```python 
from urllib.parse import urljoin, parse_qs, unquote
```
[Unquote](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.unquote) removes the spaces added by google query engine. Also, I've not seen where the link returns the link as  `url` param. So it's safe to assume it's only as the `q` param.
```python
		return unquote(url_qs['q'][0])
    return url      
```

Return the URL as the default return

Reply:
@MeNsaaH It's not safe, I have seen instances where the correct value was in `'url'` (though it was with `hl=...` different from `'en'`
Besides, it's never safe to do `url_qs['q']`, the '[' operator throws an exception if `'q'` is not there.

I made the other changes you mentioned.